internal/credentials/spiffe/spiffe.go

```diff
@@ -40,27 +44,38 @@ type partialParsedSPIFFEBundleMap struct {
 // If duplicate keys are encountered in the JSON parsing, Go's default unmarshal
 // behavior occurs which causes the last processed entry to be the entry in the
 // parsed map.
-func LoadSPIFFEBundleMap(filePath string) (map[string]*spiffebundle.Bundle, error) {
+func LoadSPIFFEBundleMap(filePath string) (BundleMap, error) {
 	bundleMapRaw, err := os.ReadFile(filePath)
 	if err != nil {
 		return nil, err
 	}
+	return BundleMapFromBytes(bundleMapRaw)
+}
+
+// BundleMapFromBytes parses bytes into a SPIFFE Bundle Map. See the
+// SPIFFE Bundle Map spec for more detail -
+// https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Trust_Domain_and_Bundle.md#4-spiffe-bundle-format
+// If duplicate keys are encountered in the JSON parsing, Go's default unmarshal
+// behavior occurs which causes the last processed entry to be the entry in the
+// parsed map.
+func BundleMapFromBytes(bundleMapBytes []byte) (BundleMap, error) {
 	var result partialParsedSPIFFEBundleMap
-	if err := json.Unmarshal(bundleMapRaw, &result); err != nil {
+	err := json.Unmarshal(bundleMapBytes, &result)
+	if err != nil {
 		return nil, err
 	}
 	if result.Bundles == nil {
-		return nil, fmt.Errorf("spiffe: no content in spiffe bundle map file %v", filePath)
+		return nil, fmt.Errorf("spiffe: BundleMapFromBytes() no bundles parsed from spiffe bundle map bytes")
 	}
 	bundleMap := map[string]*spiffebundle.Bundle{}
 	for td, jsonBundle := range result.Bundles {
 		trustDomain, err := spiffeid.TrustDomainFromString(td)
 		if err != nil {
-			return nil, fmt.Errorf("spiffe: invalid trust domain (%v) found when parsing map file (%v): %v", td, filePath, err)
+			return nil, fmt.Errorf("spiffe: BundleMapFromBytes() invalid trust domain (%v) found when parsing SPIFFE Bundle Map: %v", td, err)
```

Comment:
Nit/Optional: Here and in the next error returned down below, you could consider using a `%q` formatting directive to output the trust domain. It will appear as a quoted string in the returned error, which is nicer than having to surround it with parentheses.

Reply:
Done, thanks for the tip!